python/res/fm/ecl/ecl_run.py

```diff
@@ -0,0 +1,385 @@
+import os.path
+import re
+import time
+import datetime
+import socket
+from collections import namedtuple
+
+try:
+    from ecl.summary import EclSum
+except ImportError:
+    from ert.ecl import EclSum
+
+from .ecl_config import EclConfig
+
+
+EclipseResult = namedtuple("EclipseResult" , "errors bugs")
+body_sub_pattern = "(\s^\s@.+$)*"
+date_sub_pattern = "\s+AT TIME\s+(?P<Days>\d+\.\d+)\s+DAYS\s+\((?P<Date>(.+)):\s*$"
+error_pattern    = "^\s@--  ERROR%s$%s"   % (date_sub_pattern, body_sub_pattern)
+
+def make_LSB_MCPU_machine_list( LSB_MCPU_HOSTS ):
+    host_numcpu_list = LSB_MCPU_HOSTS.split()
+    host_list = []
+    for index in range(len(host_numcpu_list) / 2):
+        machine = host_numcpu_list[ 2*index ]
+        host_numcpu = int(host_numcpu_list[ 2*index  + 1 ])
+        for icpu in range(host_numcpu):
+            host_list.append( machine )
+    return host_list
+
+
+
+def make_LSB_machine_list( LSB_HOSTS ):
+    return LSB_HOSTS.split()
+
+
+
+
+class EclRun(object):
+    """Wrapper class to run Eclipse simulations.
+
+    The EclRun class is a small wrapper class which is used to run Eclipse
+    simulations. It will load a configuration, i.e. where the binary is
+    installed and so on, from an instance of the EclConfig class.
+
+    The main method is the runEclipse() method which will:
+
+      1. Set up redirection of the stdxxx file descriptors.
+      2. Set the necessary environment variables.
+      3. [MPI]: Create machine_file listing the nodes which should be used.
+      4. fork+exec to actually run the Eclipse binary.
+      5. Parse the output .PRT / .ECLEND file to check for errors.
+
+    If the simulation fails the runEclipse() method will raise an exception.
+
+    The class is called EclRun, and the main focus has been on running Eclipse
+    simulations, but it should also handle "eclipse-like" simulators, e.g. the
+    simulator OPM/flow.
+
+    To actually create an executable script based on this class could in it's
+    simplest form be:
+
+       #!/usr/bin/env python
+       import sys
+       from res.fm.ecl import EclRun
+
+       run = EclRun(sys.argv)
+       run.runEclipse( )
+
+
+    Name of executable
+    ==================
+
+    Observe that the EclRun.__init__() initializer determines which simulator
+    to use by checking the value of sys.argv[0] - i.e. typically the name of
+    the executable. The currently recognized names are:
```

Comment:
I know it was like this before and it works, but I honestly find this quite ugly.  Can we consider changing the constructor of EclRun to something like `def __init__(self, simulator, version, data_file, num_cpu = 1)` ?